content/posts/2022-11-02-how-to-fix-django-admin-timeouts/index.md

```diff
@@ -0,0 +1,66 @@
+---
+title: How to fix Django admin timeouts
+author: Luiz Lima
+twitter: umluizlima
+layout: post
+lang: en
+path: /blog/how-to-fix-django-admin-timeouts
+date: '2022-11-02'
+comments: true
+---
+
+Have you ever noticed some Django admin page taking a long time to load? Maybe even ending in a timeout (http 504)?
+
+![Example](./example.png)
+
+This kind of issue should be uncommon in small, earlier projects. However, if it starts to happen and gets worse over time, one possible cause is that some of that entity’s relationships are editable on the page, which causes Django to fetch all instances in the database to populate the dropdown list.
+
+### Use `readonly_fields`
+
+In this situation, the easy solution would be to just set those fields as `readonly_fields` in the page definition, like [this](https://docs.djangoproject.com/en/4.1/ref/contrib/admin/#django.contrib.admin.ModelAdmin.readonly_fields).
```

Comment:
I would avoid using "this" and "here" as the call-to-action, because of SEO/indexing purposes:
 
```suggestion
In this situation, the easiest solution would be to just set those fields as [`readonly_fields`](https://docs.djangoproject.com/en/4.1/ref/contrib/admin/#django.contrib.admin.ModelAdmin.readonly_fields) in the page definition.
```